src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -384,7 +384,7 @@ public static void firePlayerSavingEvent(EntityPlayer player, File playerDirecto
     public static void firePlayerLoadingEvent(EntityPlayer player, IPlayerFileData playerFileData, String uuidString)
     {
         SaveHandler sh = (SaveHandler) playerFileData;
-        File dir = ObfuscationReflectionHelper.getPrivateValue(SaveHandler.class, sh, "playersDirectory", "field_"+"75771_c");
```

Comment:
There is a reason why the srg names are split into two strings.. can you guess?

Reply:
```
* @param methodName     The name of the method to find (used in developer environments, i.e. "getWorldTime").
* @param methodObfName  The obfuscated name of the method to find (used in obfuscated environments, i.e. "getWorldTime").
```

Reply:
JLS seems to suggest it should be computed at compile time.
`Strings computed by constant expressions (§15.28) are computed at compile time and then treated as if they were literals.`

This should still be a compile time constant expression e.g.
`"The integer " + Long.MAX_VALUE + " is mighty big."` is given as an example of a constant expression here: https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.28

Reply:
Possible, but it still looks better the old way.